### ISSUE TITLE
Workload metrics tab not displayed

### ISSUE THREAD

**Issue by bla-ckbox** [rancher/rancher#36587]
**Rancher Server Setup**
- Rancher version: V2.6.3
- Installation option : Helm Chart on RKE1 cluster

**Information about the Cluster**
- Kubernetes version: v1.20.11
- Cluster Type: Downstream
   - Custom (RKE1)

**User Information**
The basic permission is User-Base (login-access only).
User have a project role inspired by the View Workloads and View Config Maps role and which applyed to the project containing the workloads .
This is complemented by a second role that inherits the View Monitoring and Project Monitoring View Role and which is applied to the System project containing cattle-monitoring-system namespace .

**Describe the bug**
User cant see  the metrics tab on the workloads page.

**To Reproduce**
Create User with *View Workloads* role on a project and with minimu project role to see/access to the monitoring

**Result**
User can see/access to the monitoring
But workload-metrics tab not displayed when wiewing a workload (daemonset or statefullset for example)

**Screenshots**
![pod-no-metrics](https://user-images.githubusercontent.com/2076045/154799817-04b8bdf2-d89a-4629-919b-861ab8e39c95.png)

**Expected Result**
User can see  the metrics tab on the workloads page, for example here is what I see with my cluster owner profile:
![pod-tab-metrics](https://user-images.githubusercontent.com/2076045/154799951-74a6c9b9-4c5e-49a0-b094-189778123753.png)

**Additional context**
See https://forums.rancher.com/t/advanced-view-workloads-role/37032

**Comment by bla-ckbox** [comment]
I finally managed to bring up the metrics tab, it depends on the following permission:

```
- apiGroups:
  - catalog.cattle.io
  resources:
  - apps
  verbs:
  - list
```

I also solved another problem, because despite the tab being visible, an error popin appeared a few seconds after being on a page with metrics (this also applies to Grafana).

![metrics-popin-error](https://user-images.githubusercontent.com/2076045/154836558-ed1e8261-2af6-406d-992d-a69df2e97d78.png)

This is from a POST call on _frontend-metrics_ url:
![post403](https://user-images.githubusercontent.com/2076045/154836706-d1829bc0-979f-4244-8aa7-1616ad27657b.png)

I got around this by applying

```
- apiGroups:
  - '*'
  resources:
  - services/proxy
  verbs:
  - create
```

To summarize my role template to see the metrics gives this:
```
roleTemplateNames:
- monitoring-ui-view
- project-monitoring-readonly
rules:
- apiGroups:
  - catalog.cattle.io
  resources:
  - apps
  verbs:
  - list
- apiGroups:
  - '*'
  resources:
  - services/proxy
  verbs:
  - create
```

**Comment by kpanic9** [comment]
I'm not able to see Metrics tab for an imported EKS cluster even with the Admin role. Did you have to do anything special to get it visible to Cluster Owner?

**Comment by fgielow** [comment]
I am experiencing the same issue after upgrading from rancher `2.5.7` to `2.6.3` and upgrading the `rancher-monitoring` chart from `9.4.203` to `100.0.0+up16.6.0`.

**Comment by fgielow** [comment]
> I finally managed to bring up the metrics tab, it depends on the following permission:
> 
> ```
> - apiGroups:
>   - catalog.cattle.io
>   resources:
>   - apps
>   verbs:
>   - list
> ```
> 
> I also solved another problem, because despite the tab being visible, an error popin appeared a few seconds after being on a page with metrics (this also applies to Grafana).
> 
> ![metrics-popin-error](https://user-images.githubusercontent.com/2076045/154836558-ed1e8261-2af6-406d-992d-a69df2e97d78.png)
> 
> This is from a POST call on _frontend-metrics_ url: ![post403](https://user-images.githubusercontent.com/2076045/154836706-d1829bc0-979f-4244-8aa7-1616ad27657b.png)
> 
> I got around this by applying
> 
> ```
> - apiGroups:
>   - '*'
>   resources:
>   - services/proxy
>   verbs:
>   - create
> ```
> 
> To summarize my role template to see the metrics gives this:
> 
> ```
> roleTemplateNames:
> - monitoring-ui-view
> - project-monitoring-readonly
> rules:
> - apiGroups:
>   - catalog.cattle.io
>   resources:
>   - apps
>   verbs:
>   - list
> - apiGroups:
>   - '*'
>   resources:
>   - services/proxy
>   verbs:
>   - create
> ```

Can confirm adding the create permission to services/proxy also has worked here.

**Comment by brgsousa** [comment]
I am using rancher 2.7.1, tried those roles but didn't work

**Comment by fbrokanlou** [comment]
> I am using rancher 2.7.5 (and monitoring:102.0.1+up40.1.2), it worked using like this:
> 
> ```
> rules:
>   - apiGroups:
>       - catalog.cattle.io
>     resources:
>       - apps
>     verbs:
>       - list
>       - get
>   - apiGroups:
>       - '*'
>     resources:
>       - services/proxy
>     verbs:
>       - create
> ```

Hi, I am using the same Rancher and monitoring version (`rancher 2.7.5 (and monitoring:102.0.1+up40.1.2)`). which role did you edit for this to work, please?

**Comment by hhstu** [comment]
> 我正在使用 rancher 2.7.5 （和监控：102.0.1+up40.1.2），它的使用方式如下：
> 
> ```
> rules:
>   - apiGroups:
>       - catalog.cattle.io
>     resources:
>       - apps
>     verbs:
>       - list
>       - get
>   - apiGroups:
>       - '*'
>     resources:
>       - services/proxy
>     verbs:
>       - create
> ```

is this roletemplates ? @brgsousa

**Comment by brgsousa** [comment]
These are the rules. You have to apply them to your roletemplate

**Comment by hhstu** [comment]
Thanks for your reply，  I fock project-member roletemplate, and add services/proxy ,but it not work, can you help me check it ?  @brgsousa 
```yaml
apiVersion: management.cattle.io/v3
builtin: false
context: project
description: ""
displayName: project-fock
external: false
hidden: false
kind: RoleTemplate
metadata:
  annotations:
    cleanup.cattle.io/rtUpgradeCluster: "true"
    field.cattle.io/creatorId: u-y5jsklefrd
    lifecycle.cattle.io/create.mgmt-auth-roletemplate-lifecycle: "true"
  creationTimestamp: "2023-10-23T07:37:05Z"
  finalizers:
  - controller.cattle.io/mgmt-auth-roletemplate-lifecycle
  generateName: rt-
  generation: 3
  labels:
    cattle.io/creator: norman
  name: rt-wfxcv
  resourceVersion: "46951042"
  uid: 375df78b-c7af-4858-86c0-0aa2d5ecf8a8
roleTemplateNames:
- edit
rules:
- apiGroups:
  - ui.cattle.io
  resources:
  - navlinks
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectroletemplatebindings
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - project.cattle.io
  resources:
  - apps
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - project.cattle.io
  resources:
  - apprevisions
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - namespaces
  verbs:
  - create
- apiGroups:
  - ""
  resources:
  - persistentvolumes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - storage.k8s.io
  resources:
  - storageclasses
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - apiregistration.k8s.io
  resources:
  - apiservices
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - persistentvolumeclaims
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - metrics.k8s.io
  resources:
  - pods
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - clusterevents
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - notifiers
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectalertrules
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectalertgroups
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectloggings
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - clustercatalogs
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectcatalogs
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - projectmonitorgraphs
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - catalogtemplates
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resources:
  - catalogtemplateversions
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - monitoring.cattle.io
  resources:
  - prometheus
  verbs:
  - view
- apiGroups:
  - monitoring.coreos.com
  resources:
  - prometheuses
  - prometheusrules
  - servicemonitors
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - networking.istio.io
  resources:
  - destinationrules
  - envoyfilters
  - gateways
  - serviceentries
  - sidecars
  - virtualservices
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - config.istio.io
  resources:
  - apikeys
  - authorizations
  - checknothings
  - circonuses
  - deniers
  - fluentds
  - handlers
  - kubernetesenvs
  - kuberneteses
  - listcheckers
  - listentries
  - logentries
  - memquotas
  - metrics
  - opas
  - prometheuses
  - quotas
  - quotaspecbindings
  - quotaspecs
  - rbacs
  - reportnothings
  - rules
  - solarwindses
  - stackdrivers
  - statsds
  - stdios
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - authentication.istio.io
  resources:
  - policies
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - rbac.istio.io
  resources:
  - rbacconfigs
  - serviceroles
  - servicerolebindings
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - security.istio.io
  resources:
  - authorizationpolicies
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - catalog.cattle.io
  resources:
  - clusterrepos
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - catalog.cattle.io
  resources:
  - operations
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - catalog.cattle.io
  resources:
  - releases
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - catalog.cattle.io
  resources:
  - apps
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - management.cattle.io
  resourceNames:
  - local
  resources:
  - clusters
  verbs:
  - get
- apiGroups:
  - catalog.cattle.io
  resources:
  - apps
  verbs:
  - list
  - get
- apiGroups:
  - '*'
  resources:
  - services/proxy
  verbs:
  - '*'
```

**Comment by brgsousa** [comment]
> I am using rancher 2.7.5 (and monitoring:102.0.1+up40.1.2), it worked using like this:
> 
> ```
> rules:
>   - apiGroups:
>       - catalog.cattle.io
>     resources:
>       - apps
>     verbs:
>       - list
>       - get
>   - apiGroups:
>       - '*'
>     resources:
>       - services/proxy
>     verbs:
>       - create
> ```

Besides adding those rules, I have inherited these roles:
Manage Workloads
View Services
View Volumes
View Monitoring
View Nodes
Manage Secrets
Manage Config Maps
Kubernetes edit
Project Monitoring View Role

**Comment by hhstu** [comment]
@brgsousa Thanks for your reply, success is create an cluster role,not a project role
